content/swiftui/swiftui.md

```diff
@@ -0,0 +1,24 @@
+---
+Title: 'SwiftUI'
+Description: 'SwiftUI is a framework used to build user interfaces for applications within the Apple ecosystem.'
+Subjects:
+  - 'Mobile Development'
+  - 'Computer Science'
+Tags:
+  - 'SwiftUI'
+  - 'UI'
+  - 'UX'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**SwiftUI** was released in 2019 and is a framework used to build user interfaces for iOS 13+, macOS, iPadOS, tvOS, and watchOS applications.
+
+The framework provides views, controls, and layout structures through declarative programming. This programming paradigm means that the developer declares how an application should function and appear in different states, but the actual changes in state are managed by the application itself.
```

Comment:
I mentioned programming paradigm here but realize that it's not a general entry concept. Should we maybe consider adding that as one and nesting the paradigms underneath as terms?

@Dusch4593 we discussed organizing the general section previously but thinking it's more of a possible action item now. I can put a PR for this. Can I also get an approval by someone else on free team if this is a good idea? @SSwiniarski @KTom101 @HishamT

Reply:
Sounds good to me.

Reply:
+1

Reply:
@SSwiniarski @Dusch4593 I created the [PR 1037]( https://github.com/Codecademy/docs/pull/1037) for this, as discussed. Please let me know if this looks good and I can also put out more issues for programming paradigms!